### DISCUSSION TITLE
How to use swc_core to enable swc/plugin features

### DISCUSSION THREAD

**Discussion by xin2017338** [swc-project/swc#7502] [Q&A] [answered]
I want to write a rust version of SWC-Loader, but I don't know how to turn on the swc/plugin feature, could you give me some guidance

I want to enable this functionality in swc-core and then inject it through the serde json method

@kdy1 thank you
My dependence
```
[dependencies]
anyhow               = { workspace = true }
async-trait          = { workspace = true }
once_cell            = { workspace = true }
rspack_core          = { path = "../rspack_core" }
rspack_error         = { path = "../rspack_error" }
rspack_loader_runner = { path = "../rspack_loader_runner" }
serde                = { workspace = true, features = ["derive"] }
str_indices          = "0.4.1"
swc_config           = { workspace = true }
swc_core             = { workspace = true }
tokio                = { workspace = true, features = ["rt", "rt-multi-thread", "macros", "test-util", "parking_lot"] }

```

config
```
      jsc: JscConfig {
        experimental: JscExperimental {
          plugins: Some(vec![PluginConfig(
            tests_path
              .join("my_first_plugin.wasm")
              .as_os_str()
              .to_string_lossy()
              .to_string(),
            serde_json::Value::Bool(true),
          )]),
          keep_import_assertions: Default::default(),
          cache_root: None,
        },
```

**Comment by kwonoj** [comment, marked as answer]
`enabling` doesn't mean it'll load plugin automatically. You have to use those enabled feature and actually read & run wasm binaries.

**Reply by xin2017338** [reply]
I'm sorry that I don't understand what you mean. I want to implement this function through swc_core dependency. I tried to add plugin_transform_host_native feature, but it didn't work properly
```
   anyhow::Context::context(
            c.process_js_with_custom_pass(
              fm,
              None,
              handler,
              &options,
              comments,
              |_a| noop(),
              |_a| noop(),
            ),
            "failed to process js file",
          )
```

**Reply by xin2017338** [reply]
https://github.com/swc-project/swc/blob/main/crates/binding_macros/Cargo.toml
I see that wasm under the binding_macros module also does not have feature turned on but that feature is available

**Reply by kwonoj** [reply]
`enabling` a feature means it exposes necessary api to construct an implementation. swc_core does not provide any implementation can actually run plugin and it's up to you how / what runtime (wasmtime, v8..) to actually execute it.

**Reply by xin2017338** [reply]
Thank you. I've solved the problem
